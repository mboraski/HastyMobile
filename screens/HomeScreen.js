// Third Party Imports
import React, { Component } from 'react';
import {
    ScrollView,
    StyleSheet,
    Text,
    View,
    Image,
    TouchableOpacity,
    Platform
} from 'react-native';
import { connect } from 'react-redux';
import { MaterialIcons } from '@expo/vector-icons';

// Relative Imports
import ProductList from '../components/ProductList';
import MenuButton from '../components/MenuButton';
import CartButton from '../components/CartButton';
import SearchBar from '../components/SearchBar';
import Color from '../constants/Color';
import Style from '../constants/Style';
import { addToCart } from '../actions/cartActions';
import { selectDeliveryType } from '../actions/productActions';
import { getProductsByDeliveryType } from '../selectors/productSelectors';
import { emY } from '../utils/em';

const SOURCE = { uri: 'https://source.unsplash.com/random/800x600' };

const FILTERS = [{ name: 'For You', id: '1' }, { name: 'Food', id: '2' }];

class HomeScreen extends Component {
    callAddToCart = product => {
        this.props.addToCart(product);
    };

    goToCheckout = () => {
        this.props.navigation.navigate('cart');
    };

    renderFilter = filter => {
        const selectedFilter = this.props.deliveryType === filter.id;
        const filterButtonSelected = selectedFilter ? styles.filterButtonSelected : null;
        const filterButtonTextSelected = selectedFilter ? styles.filterButtonTextSelected : null;
        const onPress = () => this.props.selectFilter(filter.id);
        return (
            <TouchableOpacity
                key={filter.id}
                style={[styles.filterButton, filterButtonSelected]}
                onPress={onPress}
            >
                <Text style={[styles.filterButtonText, filterButtonTextSelected]}>
                    {filter.name}
                </Text>
            </TouchableOpacity>
        );
    };

    render() {
        const { cart, products } = this.props;
        return (
            <View style={styles.container}>
                {cart.totalOrders > 0 ? (
                    <TouchableOpacity style={styles.checkout} onPress={this.goToCheckout}>
                        <Text style={styles.imageTitle}>Go to Checkout</Text>
                        <View style={styles.checkoutIconContainer}>
                            <MaterialIcons
                                name="keyboard-arrow-right"
                                color="#fff"
                                size={50}
                                style={styles.checkoutIcon}
                            />
                        </View>
                    </TouchableOpacity>
                ) : (
                    <Image source={SOURCE} style={styles.image}>
                        <View style={[StyleSheet.absoluteFill, styles.imageTint]} />
                        <View>
                            <Text style={styles.imageTitle}>Recommended</Text>
                            <Text style={styles.imageMeta}>215 items</Text>
                        </View>
                    </Image>
                )}
                <ScrollView
                    horizontal
                    style={styles.filters}
                    contentContainerStyle={styles.filtersContent}
                >
                    {FILTERS.map(this.renderFilter)}
                </ScrollView>
                <ProductList cart={cart} products={products} callAddToCart={this.callAddToCart} />
            </View>
        );
    }
}
// {this.renderProducts()}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff'
    },
    image: {
        height: emY(12.5),
        alignItems: 'center',
        justifyContent: 'center'
    },
    imageTint: {
        backgroundColor: 'rgba(0, 0, 0, 0.5)'
    },
    imageTitle: {
        color: 'white',
        backgroundColor: 'transparent',
        fontSize: emY(1.875),
        textAlign: 'center'
    },
    imageMeta: {
        color: 'white',
        backgroundColor: 'transparent',
        fontSize: emY(1),
        textAlign: 'center'
    },
    checkout: {
        height: emY(12.5),
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: Color.GREEN_500
    },
    checkoutIconContainer: {
        position: 'absolute',
        top: 0,
        right: 0,
        bottom: 0,
        width: 50,
        justifyContent: 'center'
    },
    checkoutIcon: {
        backgroundColor: 'transparent'
    },
    filters: {
        ...Platform.select({
            ios: {
                height: emY(6.25)
            },
            android: {
                height: emY(6.25) + 30
            }
        })
    },
    filtersContent: {
        paddingLeft: 15,
        alignItems: 'center',
        justifyContent: 'center'
    },
    filterButton: {
        minWidth: 130,
        height: emY(2.6875),
        borderColor: Color.GREY_400,
        borderWidth: StyleSheet.hairlineWidth * 2,
        borderRadius: 50,
        paddingVertical: emY(0.75),
        marginRight: 15,
        alignItems: 'center',
        justifyContent: 'center'
    },
    filterButtonSelected: {
        borderColor: Color.GREY_300,
        backgroundColor: Color.GREY_300
    },
    filterButtonText: {
        color: Color.GREY_400,
        fontSize: emY(1.125)
    },
    filterButtonTextSelected: {
        color: '#fff'
    }
});

HomeScreen.navigationOptions = {
    title: 'Hasty Logo',
    headerLeft: <MenuButton />,
    headerTitle: <SearchBar />,
    headerRight: <CartButton />,
    headerStyle: Style.headerLarge,
    headerTitleStyle: Style.headerTitle
};

const mapStateToProps = state => ({
    cart: state.cart,
    products: getProductsByDeliveryType(state),
    deliveryType: state.product.deliveryType
});

const mapDispatchToProps = dispatch => ({
    selectFilter: filter => dispatch(selectDeliveryType(filter)),
    addToCart: product => dispatch(addToCart(product))
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
