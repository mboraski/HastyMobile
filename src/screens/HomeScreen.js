// Third Party Imports
import React, { Component } from 'react';
import {
    ActivityIndicator,
    ScrollView,
    StyleSheet,
    View,
    ImageBackground,
    TouchableOpacity,
    Platform
} from 'react-native';
import { connect } from 'react-redux';
// eslint-disable-next-line import/no-extraneous-dependencies
import { MaterialIcons } from '@expo/vector-icons';
import _ from 'lodash';
// import firebase from '../firebase';

// Relative Imports
import ProductList from '../components/ProductList';
// import MenuButton from '../components/MenuButton';
// import CartButton from '../components/CartButton';
// import SearchBar from '../components/SearchBar';
import Text from '../components/Text';
import Color from '../constants/Color';
import Dimensions from '../constants/Dimensions';
// import Style from '../constants/Style';
import { addToCart } from '../actions/cartActions';
import { selectCategory, fetchProductsRequest } from '../actions/productActions';
import { dropdownAlert } from '../actions/uiActions';
import getProductsState from '../selectors/productSelectors';
import { emY } from '../utils/em';
import AuthScreenBackground from '../assets/AuthScreenBackground.jpg';
import MapHeader from '../containers/MapHeader';


class HomeScreen extends Component {
    componentDidMount() {
        this.props.fetchProductsRequest();
        if (this.props.itemCountUp) {
            this.props.dropdownAlert(true, 'More products available!');
        } else if (this.props.itemCountDown) {
            this.props.dropdownAlert(true, 'Some products are no longer available');
        }
    }

    componentWillReceiveProps(nextProps) {
        if (this.props.header.toggleState !== nextProps.header.toggleState) {
            if (nextProps.header.isMenuOpen) {
                this.props.navigation.navigate('DrawerOpen');
            } else {
                this.props.navigation.navigate('DrawerClose');
            }
        }
        if (!this.props.cart && nextProps.cart) {
            this.props.fetchProductsRequest();
        }
        if (!this.props.itemCountUp && nextProps.itemCountUp) {
            this.props.dropdownAlert(true, 'More products available!');
        } else if (!this.props.itemCountDown && nextProps.itemCountDown) {
            this.props.dropdownAlert(true, 'Some products are no longer available');
        } else {
            this.props.dropdownAlert(false, '');
        }
    }

    callAddToCart = (product) => {
        this.props.addToCart(product);
    };

    formatCategory = (category) =>
        (category ? category.toUpperCase() : '')

    goToCheckout = () => {
        this.props.navigation.navigate('checkout');
    };

    renderCategories = () => {
        const selectedCategory = this.props.category;
        return _.map(this.props.categories, (category, i) => {
            const selectedFilter = category.toLowerCase() === selectedCategory.toLowerCase();
            const filterButtonSelected = selectedFilter ? styles.filterButtonSelected : null;
            const filterButtonTextSelected = selectedFilter ?
                styles.filterButtonTextSelected : null;
            const onPress = () => this.props.selectFilter(category);
            return (
                <TouchableOpacity
                    key={i}
                    style={[styles.filterButton, filterButtonSelected]}
                    onPress={onPress}
                >
                    <Text style={[styles.filterButtonText, filterButtonTextSelected]}>
                        {category}
                    </Text>
                </TouchableOpacity>
            );
        });
    };

    render() {
        const {
            cartQuantity,
            productPending,
            productsShown,
            category,
            numberOfProducts,
        } = this.props;
        return (
            <View style={styles.container}>
                {cartQuantity > 0 ? (
                    <TouchableOpacity style={styles.checkout} onPress={this.goToCheckout}>
                        <Text style={styles.imageTitle}>Go to Checkout</Text>
                        <View style={styles.checkoutIconContainer}>
                            <MaterialIcons
                                name="keyboard-arrow-right"
                                color="#fff"
                                size={50}
                                style={styles.checkoutIcon}
                            />
                        </View>
                    </TouchableOpacity>
                ) : (
                    <ImageBackground source={AuthScreenBackground} style={styles.image}>
                        <View style={[StyleSheet.absoluteFill, styles.imageTint]} />
                        <View>
                            <Text style={styles.imageTitle}>{this.formatCategory(category)}</Text>
                            <Text style={styles.imageMeta}>
                                {numberOfProducts} items
                            </Text>
                        </View>
                    </ImageBackground>
                )}
                {productPending ? (
                    <ActivityIndicator
                        size="large"
                        style={StyleSheet.absoluteFill}
                    />
                ) :
                    <View>
                        <ScrollView
                            horizontal
                            style={styles.filters}
                            contentContainerStyle={styles.filtersContent}
                        >
                            {this.renderCategories()}
                        </ScrollView>
                        {productsShown &&
                            <ProductList
                                products={productsShown}
                                callAddToCart={this.callAddToCart}
                            />
                        }
                    </View>
                }
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#fff'
    },
    image: {
        height: Dimensions.window.height / 4,
        alignItems: 'center',
        justifyContent: 'center'
    },
    imageTint: {
        backgroundColor: 'rgba(0, 0, 0, 0.5)'
    },
    imageTitle: {
        color: 'white',
        backgroundColor: 'transparent',
        fontSize: emY(1.875),
        textAlign: 'center'
    },
    imageMeta: {
        color: 'white',
        backgroundColor: 'transparent',
        fontSize: emY(1),
        textAlign: 'center'
    },
    checkout: {
        height: Dimensions.window.height / 4,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: Color.GREEN_500
    },
    checkoutIconContainer: {
        position: 'absolute',
        top: 0,
        right: 0,
        bottom: 0,
        width: 50,
        justifyContent: 'center'
    },
    checkoutIcon: {
        backgroundColor: 'transparent'
    },
    filters: {
        ...Platform.select({
            ios: {
                height: emY(6.25)
            },
            android: {
                height: emY(6.25) + 30
            }
        })
    },
    filtersContent: {
        paddingLeft: 15,
        alignItems: 'center',
        justifyContent: 'center'
    },
    filterButton: {
        minWidth: 120,
        height: emY(2.5),
        borderColor: Color.GREY_400,
        borderWidth: StyleSheet.hairlineWidth * 2,
        borderRadius: 50,
        paddingVertical: emY(0.75),
        marginRight: 15,
        alignItems: 'center',
        justifyContent: 'center'
    },
    filterButtonSelected: {
        borderColor: Color.GREY_300,
        backgroundColor: Color.GREY_300
    },
    filterButtonText: {
        color: Color.GREY_400,
        fontSize: emY(1.125)
    },
    filterButtonTextSelected: {
        color: '#fff'
    }
});

HomeScreen.navigationOptions = {
    header: <MapHeader />
};

// HomeScreen.navigationOptions = ({ navigation }) => ({
//     title: 'Hasty Logo',
//     headerLeft: <MenuButton />,
//     headerTitle: <SearchBar />,
//     headerRight: <CartButton navigation={navigation} />,
//     headerStyle: Style.headerLarge,
//     headerTitleStyle: Style.headerTitle
// });

const mapStateToProps = state => getProductsState(state);

const mapDispatchToProps = dispatch => ({
    selectFilter: category => dispatch(selectCategory(category)),
    addToCart: productInfo => dispatch(addToCart(productInfo)),
    fetchProductsRequest: () => dispatch(fetchProductsRequest()),
    dropdownAlert: (visible, message) => dispatch(dropdownAlert(visible, message))
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
