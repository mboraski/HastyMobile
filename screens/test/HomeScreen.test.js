import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import sinon from 'sinon';

import HomeScreen from '../HomeScreen';

const initialState = {
    product: {
        list: {
            1: {
                1: {
                    quantity: 1,
                    title: 'Type 1 code 1',
                    thumbnail_image: 'image',
                    price: '$3.49'
                }
            },
            2: {
                2: {
                    quantity: 2,
                    title: 'Type 2 code 2',
                    thumbnail_image: 'image',
                    price: '$3.49'
                }
            }
        },
        deliveryType: '1'
    },
    cart: {
        totalProducts: 0
    }
};

describe('HomeScreen', () => {
    const middlewares = [];
    const mockStore = configureStore(middlewares);
    it('renders correctly', () => {
        const wrapper = shallow(<HomeScreen />, {
            context: { store: mockStore(initialState) }
        });
        const render = wrapper.dive();
        expect(render).toMatchSnapshot();
    });
});
