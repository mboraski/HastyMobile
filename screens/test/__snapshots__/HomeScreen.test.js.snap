// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "backgroundColor": "#fff",
            "flex": 1,
          }
    }
>
    <Image
        source={
            Object {
                "uri": "https://source.unsplash.com/random/800x600",
              }
        }
        style={
            Object {
                "alignItems": "center",
                "height": 200,
                "justifyContent": "center",
              }
        }
    >
        <View
            style={
                Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Object {
                      "backgroundColor": "rgba(0, 0, 0, 0.5)",
                    },
                  ]
            }
        />
        <View>
            <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                    Object {
                        "backgroundColor": "transparent",
                        "color": "white",
                        "fontSize": 30,
                        "textAlign": "center",
                      }
                }
            >
                Recommended
            </Text>
            <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                    Object {
                        "backgroundColor": "transparent",
                        "color": "white",
                        "fontSize": 16,
                        "textAlign": "center",
                      }
                }
            >
                215 items
            </Text>
        </View>
    </Image>
    <ScrollViewMock
        contentContainerStyle={
            Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingLeft": 15,
              }
        }
        horizontal={true}
        style={
            Object {
                "height": 100,
              }
        }
    >
        <TouchableOpacity
            activeOpacity={0.2}
            focusedOpacity={0.7}
            onPress={[Function]}
            style={
                Array [
                    Object {
                      "alignItems": "center",
                      "borderColor": "#b2b2b2",
                      "borderRadius": 50,
                      "borderWidth": 1,
                      "height": 43,
                      "justifyContent": "center",
                      "marginRight": 15,
                      "minWidth": 130,
                      "paddingVertical": 12,
                    },
                    Object {
                      "backgroundColor": "#cccccc",
                      "borderColor": "#cccccc",
                    },
                  ]
            }
        >
            <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                    Array [
                        Object {
                          "color": "#b2b2b2",
                          "fontSize": 18,
                        },
                        Object {
                          "color": "#fff",
                        },
                      ]
                }
            >
                For You
            </Text>
        </TouchableOpacity>
        <TouchableOpacity
            activeOpacity={0.2}
            focusedOpacity={0.7}
            onPress={[Function]}
            style={
                Array [
                    Object {
                      "alignItems": "center",
                      "borderColor": "#b2b2b2",
                      "borderRadius": 50,
                      "borderWidth": 1,
                      "height": 43,
                      "justifyContent": "center",
                      "marginRight": 15,
                      "minWidth": 130,
                      "paddingVertical": 12,
                    },
                    null,
                  ]
            }
        >
            <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                    Array [
                        Object {
                          "color": "#b2b2b2",
                          "fontSize": 18,
                        },
                        null,
                      ]
                }
            >
                Food
            </Text>
        </TouchableOpacity>
    </ScrollViewMock>
    <ProductList
        callAddToCart={[Function]}
        cart={
            Object {
                "totalProducts": 0,
              }
        }
        products={
            Array [
                Object {
                  "price": "$3.49",
                  "productCode": "1",
                  "quantity": 1,
                  "thumbnail_image": "image",
                  "title": "Type 1 code 1",
                },
              ]
        }
    />
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "backgroundColor": "#fff",
                  "flex": 1,
                }
      }
>
      <Image
            source={
                  Object {
                        "uri": "https://source.unsplash.com/random/800x600",
                      }
            }
            style={
                  Object {
                        "alignItems": "center",
                        "height": 200,
                        "justifyContent": "center",
                      }
            }
      >
            <View
                  style={
                        Array [
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              },
                              Object {
                                "backgroundColor": "rgba(0, 0, 0, 0.5)",
                              },
                            ]
                  }
            />
            <View>
                  <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                              Object {
                                    "backgroundColor": "transparent",
                                    "color": "white",
                                    "fontSize": 30,
                                    "textAlign": "center",
                                  }
                        }
                  >
                        Recommended
                  </Text>
                  <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                              Object {
                                    "backgroundColor": "transparent",
                                    "color": "white",
                                    "fontSize": 16,
                                    "textAlign": "center",
                                  }
                        }
                  >
                        215 items
                  </Text>
            </View>
      </Image>
      <ScrollViewMock
            contentContainerStyle={
                  Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "paddingLeft": 15,
                      }
            }
            horizontal={true}
            style={
                  Object {
                        "height": 100,
                      }
            }
      >
            <TouchableOpacity
                  activeOpacity={0.2}
                  focusedOpacity={0.7}
                  onPress={[Function]}
                  style={
                        Array [
                              Object {
                                "alignItems": "center",
                                "borderColor": "#b2b2b2",
                                "borderRadius": 50,
                                "borderWidth": 1,
                                "height": 43,
                                "justifyContent": "center",
                                "marginRight": 15,
                                "minWidth": 130,
                                "paddingVertical": 12,
                              },
                              Object {
                                "backgroundColor": "#cccccc",
                                "borderColor": "#cccccc",
                              },
                            ]
                  }
            >
                  <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                              Array [
                                    Object {
                                      "color": "#b2b2b2",
                                      "fontSize": 18,
                                    },
                                    Object {
                                      "color": "#fff",
                                    },
                                  ]
                        }
                  >
                        For You
                  </Text>
            </TouchableOpacity>
            <TouchableOpacity
                  activeOpacity={0.2}
                  focusedOpacity={0.7}
                  onPress={[Function]}
                  style={
                        Array [
                              Object {
                                "alignItems": "center",
                                "borderColor": "#b2b2b2",
                                "borderRadius": 50,
                                "borderWidth": 1,
                                "height": 43,
                                "justifyContent": "center",
                                "marginRight": 15,
                                "minWidth": 130,
                                "paddingVertical": 12,
                              },
                              null,
                            ]
                  }
            >
                  <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                              Array [
                                    Object {
                                      "color": "#b2b2b2",
                                      "fontSize": 18,
                                    },
                                    null,
                                  ]
                        }
                  >
                        Food
                  </Text>
            </TouchableOpacity>
      </ScrollViewMock>
      <ProductList
            callAddToCart={[Function]}
            cart={
                  Object {
                        "totalProducts": 0,
                      }
            }
            products={
                  Array [
                        Object {
                          "price": "$3.49",
                          "productCode": "1",
                          "quantity": 1,
                          "thumbnail_image": "image",
                          "title": "Type 1 code 1",
                        },
                      ]
            }
      />
</View>,
  ],
  "options": Object {
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "_currentElement": <HomeScreen
        addToCart={[Function]}
        cart={
                Object {
                        "totalProducts": 0,
                      }
        }
        deliveryType="1"
        products={
                Array [
                        Object {
                          "price": "$3.49",
                          "productCode": "1",
                          "quantity": 1,
                          "thumbnail_image": "image",
                          "title": "Type 1 code 1",
                        },
                      ]
        }
        selectFilter={[Function]}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": HomeScreen {
        "_reactInternalInstance": [Circular],
        "callAddToCart": [Function],
        "context": Object {},
        "goToCheckout": [Function],
        "props": Object {
          "addToCart": [Function],
          "cart": Object {
            "totalProducts": 0,
          },
          "deliveryType": "1",
          "products": Array [
            Object {
              "price": "$3.49",
              "productCode": "1",
              "quantity": 1,
              "thumbnail_image": "image",
              "title": "Type 1 code 1",
            },
          ],
          "selectFilter": [Function],
        },
        "refs": Object {},
        "renderFilter": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "backgroundColor": "#fff",
                              "flex": 1,
                            }
          }
>
          <Image
                    source={
                              Object {
                                        "uri": "https://source.unsplash.com/random/800x600",
                                      }
                    }
                    style={
                              Object {
                                        "alignItems": "center",
                                        "height": 200,
                                        "justifyContent": "center",
                                      }
                    }
          >
                    <View
                              style={
                                        Array [
                                                  Object {
                                                    "bottom": 0,
                                                    "left": 0,
                                                    "position": "absolute",
                                                    "right": 0,
                                                    "top": 0,
                                                  },
                                                  Object {
                                                    "backgroundColor": "rgba(0, 0, 0, 0.5)",
                                                  },
                                                ]
                              }
                    />
                    <View>
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                                  Object {
                                                            "backgroundColor": "transparent",
                                                            "color": "white",
                                                            "fontSize": 30,
                                                            "textAlign": "center",
                                                          }
                                        }
                              >
                                        Recommended
                              </Text>
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                                  Object {
                                                            "backgroundColor": "transparent",
                                                            "color": "white",
                                                            "fontSize": 16,
                                                            "textAlign": "center",
                                                          }
                                        }
                              >
                                        215 items
                              </Text>
                    </View>
          </Image>
          <ScrollViewMock
                    contentContainerStyle={
                              Object {
                                        "alignItems": "center",
                                        "justifyContent": "center",
                                        "paddingLeft": 15,
                                      }
                    }
                    horizontal={true}
                    style={
                              Object {
                                        "height": 100,
                                      }
                    }
          >
                    <TouchableOpacity
                              activeOpacity={0.2}
                              focusedOpacity={0.7}
                              onPress={[Function]}
                              style={
                                        Array [
                                                  Object {
                                                    "alignItems": "center",
                                                    "borderColor": "#b2b2b2",
                                                    "borderRadius": 50,
                                                    "borderWidth": 1,
                                                    "height": 43,
                                                    "justifyContent": "center",
                                                    "marginRight": 15,
                                                    "minWidth": 130,
                                                    "paddingVertical": 12,
                                                  },
                                                  Object {
                                                    "backgroundColor": "#cccccc",
                                                    "borderColor": "#cccccc",
                                                  },
                                                ]
                              }
                    >
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                                  Array [
                                                            Object {
                                                              "color": "#b2b2b2",
                                                              "fontSize": 18,
                                                            },
                                                            Object {
                                                              "color": "#fff",
                                                            },
                                                          ]
                                        }
                              >
                                        For You
                              </Text>
                    </TouchableOpacity>
                    <TouchableOpacity
                              activeOpacity={0.2}
                              focusedOpacity={0.7}
                              onPress={[Function]}
                              style={
                                        Array [
                                                  Object {
                                                    "alignItems": "center",
                                                    "borderColor": "#b2b2b2",
                                                    "borderRadius": 50,
                                                    "borderWidth": 1,
                                                    "height": 43,
                                                    "justifyContent": "center",
                                                    "marginRight": 15,
                                                    "minWidth": 130,
                                                    "paddingVertical": 12,
                                                  },
                                                  null,
                                                ]
                              }
                    >
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                                  Array [
                                                            Object {
                                                              "color": "#b2b2b2",
                                                              "fontSize": 18,
                                                            },
                                                            null,
                                                          ]
                                        }
                              >
                                        Food
                              </Text>
                    </TouchableOpacity>
          </ScrollViewMock>
          <ProductList
                    callAddToCart={[Function]}
                    cart={
                              Object {
                                        "totalProducts": 0,
                                      }
                    }
                    products={
                              Array [
                                        Object {
                                          "price": "$3.49",
                                          "productCode": "1",
                                          "quantity": 1,
                                          "thumbnail_image": "image",
                                          "title": "Type 1 code 1",
                                        },
                                      ]
                    }
          />
</View>,
        "_debugID": 4,
        "_renderedOutput": <View
          style={
                    Object {
                              "backgroundColor": "#fff",
                              "flex": 1,
                            }
          }
>
          <Image
                    source={
                              Object {
                                        "uri": "https://source.unsplash.com/random/800x600",
                                      }
                    }
                    style={
                              Object {
                                        "alignItems": "center",
                                        "height": 200,
                                        "justifyContent": "center",
                                      }
                    }
          >
                    <View
                              style={
                                        Array [
                                                  Object {
                                                    "bottom": 0,
                                                    "left": 0,
                                                    "position": "absolute",
                                                    "right": 0,
                                                    "top": 0,
                                                  },
                                                  Object {
                                                    "backgroundColor": "rgba(0, 0, 0, 0.5)",
                                                  },
                                                ]
                              }
                    />
                    <View>
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                                  Object {
                                                            "backgroundColor": "transparent",
                                                            "color": "white",
                                                            "fontSize": 30,
                                                            "textAlign": "center",
                                                          }
                                        }
                              >
                                        Recommended
                              </Text>
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                                  Object {
                                                            "backgroundColor": "transparent",
                                                            "color": "white",
                                                            "fontSize": 16,
                                                            "textAlign": "center",
                                                          }
                                        }
                              >
                                        215 items
                              </Text>
                    </View>
          </Image>
          <ScrollViewMock
                    contentContainerStyle={
                              Object {
                                        "alignItems": "center",
                                        "justifyContent": "center",
                                        "paddingLeft": 15,
                                      }
                    }
                    horizontal={true}
                    style={
                              Object {
                                        "height": 100,
                                      }
                    }
          >
                    <TouchableOpacity
                              activeOpacity={0.2}
                              focusedOpacity={0.7}
                              onPress={[Function]}
                              style={
                                        Array [
                                                  Object {
                                                    "alignItems": "center",
                                                    "borderColor": "#b2b2b2",
                                                    "borderRadius": 50,
                                                    "borderWidth": 1,
                                                    "height": 43,
                                                    "justifyContent": "center",
                                                    "marginRight": 15,
                                                    "minWidth": 130,
                                                    "paddingVertical": 12,
                                                  },
                                                  Object {
                                                    "backgroundColor": "#cccccc",
                                                    "borderColor": "#cccccc",
                                                  },
                                                ]
                              }
                    >
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                                  Array [
                                                            Object {
                                                              "color": "#b2b2b2",
                                                              "fontSize": 18,
                                                            },
                                                            Object {
                                                              "color": "#fff",
                                                            },
                                                          ]
                                        }
                              >
                                        For You
                              </Text>
                    </TouchableOpacity>
                    <TouchableOpacity
                              activeOpacity={0.2}
                              focusedOpacity={0.7}
                              onPress={[Function]}
                              style={
                                        Array [
                                                  Object {
                                                    "alignItems": "center",
                                                    "borderColor": "#b2b2b2",
                                                    "borderRadius": 50,
                                                    "borderWidth": 1,
                                                    "height": 43,
                                                    "justifyContent": "center",
                                                    "marginRight": 15,
                                                    "minWidth": 130,
                                                    "paddingVertical": 12,
                                                  },
                                                  null,
                                                ]
                              }
                    >
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        style={
                                                  Array [
                                                            Object {
                                                              "color": "#b2b2b2",
                                                              "fontSize": 18,
                                                            },
                                                            null,
                                                          ]
                                        }
                              >
                                        Food
                              </Text>
                    </TouchableOpacity>
          </ScrollViewMock>
          <ProductList
                    callAddToCart={[Function]}
                    cart={
                              Object {
                                        "totalProducts": 0,
                                      }
                    }
                    products={
                              Array [
                                        Object {
                                          "price": "$3.49",
                                          "productCode": "1",
                                          "quantity": 1,
                                          "thumbnail_image": "image",
                                          "title": "Type 1 code 1",
                                        },
                                      ]
                    }
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <HomeScreen
    addToCart={[Function]}
    cart={
        Object {
            "totalProducts": 0,
          }
    }
    deliveryType="1"
    products={
        Array [
            Object {
              "price": "$3.49",
              "productCode": "1",
              "quantity": 1,
              "thumbnail_image": "image",
              "title": "Type 1 code 1",
            },
          ]
    }
    selectFilter={[Function]}
/>,
}
`;
